### DISCUSSION TITLE
Nextjs 14: Not able to add url with trailing slash in meta details like canonical url and og:url

### DISCUSSION THREAD

**Discussion by umang-goyal** [vercel/next.js#65323] [Help]
### Summary

I just migrated v13 to v14. After migrations the canonical url and og:url in metadata of the pages donot have trailing slash when I have explicitly added that. Its an issue cause the crawler will take it as different url from the previous one. 

### Example
```
export const metadata = {
  alternates: {
    canonical: 'https://nextjs.org/',
    openGraph: {
       url: 'https://nextjs.org/'
    }
}
```
`<link rel="canonical" href="https://nextjs.org">`
`<meta property="og:url" content="https://nextjs.org">`

_No response_

**Comment by Yansoul** [comment]
I encountered the same issue🥲

**Comment by HellYeahOmg** [comment]
bro same

**Comment by khasanbayev-7** [comment]
This behavior is actually expected after the migration from Next.js 13 → 14.

Starting with Next 14, `metadata` canonical and `og:url` fields are automatically **normalized without trailing slashes**, even if you manually include them.  
This is part of the new metadata handling API refactor.

If you explicitly want to preserve the trailing slash, you can define them via the `generateMetadata` function instead of static `metadata` export:

```js
export async function generateMetadata() {
  const url = 'https://nextjs.org/';
  return {
    alternates: { canonical: url },
    openGraph: { url },
  };
}

**Reply by HellYeahOmg** [reply]
Not helping... just tried on next 15.5.6, still no / in the end of the path. TrailingSlash is disabled in next config

**Reply by HellYeahOmg** [reply]
Also nothing is saying that this is expected here: https://nextjs.org/blog/next-14#metadata-improvements
